### DISCUSSION TITLE
Diff b/w internal _transfer function and public transfer function

### DISCUSSION THREAD

**Discussion by sanjay-in** [Cyfrin/foundry-full-course-cu#1783] [Q&A]
In ManualToken.sol in foundry-erc20-f23 repo, why do we have an internal _transfer function and public transfer function. The transfer function calls internal _transfer function. Cant we have a single public function which does the logic
` function _transfer(address _from, address _to, uint256 _value) internal {`
`function transfer(
        address _to,
        uint256 _value
    ) public returns (bool success) {`

**Comment by EngrPips** [comment]
Hello @sanjay-in, `ManualToken` inherited the internal `_transfer` from the `ERC20` contract, and we want to make sure people can mint `Manual` token; hence we needed an interface that would be the entrance point for users to mint our `ManualToken`. So we created a public `transfer` function that will call the internal `_transfer` function, which will then mint tokens to users. This pattern can also help with security as users have less access to the actual function that performs the activity.
Please let us know if anything is still confusing.

**Reply by sanjay-in** [reply]
Hi @EngrPips , Thanks for the clarification it's helpful!

**Reply by EngrPips** [reply]
You are welcome.
